### DISCUSSION TITLE
Using OnlyOffice

### DISCUSSION THREAD

**Discussion by italoalmeida0** [filebrowser/filebrowser#1272] [Ideas]
You could put support for OnlyOffice and in the docker in environment you could put something like this: 
```
environment:
  - ONLYOFFICE=http://onlyoffice_doc_server
```
to install the document server

**Comment by nook24** [comment]
Support for OnlyOffice would be great! I think it would need the following environment variables:
```
ONLYOFFICE_URL = 'https://example.org/web-apps/apps/api/documents/api.js'
ONLYOFFICE_FILE_EXTENSIONS = ('doc', 'docx', 'ppt', 'pptx', 'xls', 'xlsx', 'odt', 'fodt', 'odp', 'fodp', 'ods', 'fods')
ONLYOFFICE_JWT = 'SECRET'
```

**Reply by zfxkj** [reply]
Hello, I added the above code to docker run's command, but it didn't work. I only intercepted the first line of code, but there was no response. What should I do?

**Reply by nook24** [reply]
I'm sorry but this is just my suggestion of how OnlyOffice could be implemented. Unfortunately there is no feedback from the devs so far. So we don't know if Filebrowser will ever have support for OnlyOffice one day.

**Reply by zfxkj** [reply]
Thanks for your answer

**Reply by Forsskieken** [reply]
filestash does something similar

**Reply by turbcool** [reply]
I have successfully setup Filestash with openoffice, and really like it.

**Reply by ngophuong** [reply]
> I have successfully setup Filestash with openoffice, and really like it.

could you help share your docker compose file

**Comment by colin715** [comment]
Was this ever implemented? I see an open pull request for this feature back in May.

**Comment by shawnngtq** [comment]
follow

**Comment by AssKicker0214** [comment]
would be nice to see this feature

**Comment by contagon** [comment]
I would also love to see this! Curious to hear what the thought process is between including it in or not.

**Comment by ngophuong** [comment]
Definitely would be more useful and function to the application with Onlyoffice integration. Look forward to the update

**Comment by Gugu04** [comment]
it would be a high-impact update and expected by many

**Comment by pluja** [comment]
Are there any future plans on adding this?

**Comment by turbcool** [comment]
I am waiting for this feature

**Comment by x-real-ip** [comment]
Are there any future plans on adding this?

**Comment by turbcool** [comment]
Onlyoffice integration is work-in-progress #1420

**Reply by turbcool** [reply]
Somebody created a new pull request based on #1420 (as it was stale for 5 months): #2954

**Comment by defencedog** [comment]
Any news for onlyoffice or collabora office integration with filebrowser?

**Reply by colin715** [reply]
Use this fork instead: https://github.com/gtsteffaniak/filebrowser
